watchbot_progress/backends/redis.py

```diff
@@ -0,0 +1,132 @@
+from __future__ import division
+from __future__ import absolute_import
+
+import json
+import logging
+import os
+
+import boto3
+import redis
+
+from watchbot_progress.backends.base import WatchbotProgressBase
+from watchbot_progress.main import JobDoesNotExist
+
+
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.NullHandler())
+
+
+class RedisProgress(WatchbotProgressBase):
+    """Sets up objects for reduce mode job tracking with SNS and Redis
+    """
+
+    def __init__(self, topic_arn=None, host='localhost', port=6379, db=0):
+        # SNS Messages
+        self.sns = boto3.client('sns')
+        self.topic = topic_arn if topic_arn else os.environ['WorkTopic']
+
+        # Redis
+        self.redis = redis.StrictRedis(host=host, port=port, db=db)
+
+    def _metadata_key(self, jobid):
+        return '{}-metadata'.format(jobid)
+
+    def _parts_key(self, jobid):
+        return '{}-parts'.format(jobid)
+
+    def _decode_dict(self, meta):
+        return {k.decode('utf-8'): v.decode('utf-8')
+                for k, v in meta.items()}
+
+    def status(self, jobid, part=None):
+        """get status from dynamodb
+
+        Parameters
+        ----------
+        jobid: string?
+        part: optional int
+            return status of the given partid
+
+        Returns
+        -------
+        dict, similar to JS watchbot-progress.status object
+        """
+        if part is not None:
+            is_member = self.redis.sismember(self._parts_key(jobid), part)
+            return {
+                'part': part,
+                'complete': not bool(is_member)}
+
+        pipe = self.redis.pipeline()
+        pipe.hgetall(self._metadata_key(jobid))
+        pipe.scard(self._parts_key(jobid))
+        meta, remaining = pipe.execute()
+
+        meta = self._decode_dict(meta)
+        try:
+            total = int(meta['total'])
+        except KeyError:
+            raise JobDoesNotExist('Job does not exist, run set_total first')
+
+        percent = (total - remaining) / total
```

Comment:
does this return only 0 or 1 in python 2.7 ?

Reply:
It will always return a float because we're using `from __future__ import division`

Reply:
great didn't catch that 👌